src/cpu/aarch64/acl_post_ops.hpp

```diff
@@ -0,0 +1,186 @@
+
+#ifndef CPU_AARCH64_ACL_POST_OPS_HPP
+#define CPU_AARCH64_ACL_POST_OPS_HPP
+
+#include "cpu/cpu_convolution_pd.hpp"
+
+#include "cpu/aarch64/acl_binary.hpp"
+#include "cpu/aarch64/acl_eltwise.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace cpu {
+namespace aarch64 {
+
+struct acl_post_ops_t {
+
+    acl_post_ops_t() {}
+
+    // init the acl_post_ops_t. Note that this function modifies the passed in
+    // post ops by setting the preferred memory formats
+    status_t init(engine_t *engine, post_ops_t &post_ops,
+            const memory_desc_t &dst_md) {
+
+        CHECK(post_ops.set_default_formats(&dst_md));
+
+        // Reset properties derived from post_ops
+        sum_index = -1;
+        post_op_primitives = {};
+
+        for (int i = 0; i < post_ops.len(); i++) {
+            auto &po = post_ops.entry_[i];
+
+            if (po.is_sum()) {
+                ACL_CHECK_SUPPORT(po.sum.scale != 1.0f,
+                        "sum post op scale must be 1 (no scale)");
+
+                ACL_CHECK_SUPPORT(po.sum.zero_point != 0,
+                        "sum post op zero point must be 0 (no shift)");
+
+                // >= 0 means we had one already
+                ACL_CHECK_SUPPORT(sum_index >= 0,
+                        "there must not be more than 1 sum post op");
+
+                sum_index = i;
+
+                // Sum is an add primitive where dst = temp_dst + dst
+                binary_desc_t po_desc;
+                po_desc.primitive_kind = primitive_kind::binary;
+                po_desc.alg_kind = alg_kind::binary_add;
+                po_desc.src_desc[0] = dst_md;
+                po_desc.src_desc[1] = dst_md;
+                po_desc.dst_desc = dst_md;
+                auto empty_attr = dnnl_primitive_attr();
+                typename acl_binary_t::pd_t acl_binary_pd(
+                        &po_desc, &empty_attr, nullptr);
+                CHECK(acl_binary_pd.init(engine));
+
+                auto acl_binary
+                        = std::make_shared<acl_binary_t>(&acl_binary_pd);
+                CHECK(acl_binary->init(engine));
+                post_op_primitives.push_back(acl_binary);
+
+            } else if (po.is_binary()) {
+                binary_desc_t po_desc;
+                po_desc.primitive_kind = primitive_kind::binary;
+                po_desc.alg_kind = po.binary.alg;
+                po_desc.src_desc[0] = dst_md;
+                po_desc.src_desc[1] = po.binary.src1_desc;
+                po_desc.dst_desc = dst_md;
+                auto empty_attr = dnnl_primitive_attr();
+                typename acl_binary_t::pd_t acl_binary_pd(
+                        &po_desc, &empty_attr, nullptr);
+                CHECK(acl_binary_pd.init(engine));
+
+                auto acl_binary
+                        = std::make_shared<acl_binary_t>(&acl_binary_pd);
+                CHECK(acl_binary->init(engine));
+                post_op_primitives.push_back(std::move(acl_binary));
```

Comment:
QQ: why do you need to `std::move` here, but not in other places?

Reply:
Good spot, this was left over from when these were `unique_ptr`s